atlasdb-client/src/test/java/com/palantir/atlasdb/util/MetricNameUtilsTest.java

```diff
@@ -0,0 +1,136 @@
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessage("The tag value name: tag;Val1 contains the forbidden character: ;");
+    }
+
+    @Test
+    public void shouldThrowIfFirstMetricTagKeyContainsEquals() throws Exception {
+        assertThatThrownBy(() -> MetricNameUtils.getMetricName(METRIC_NAME,
+                ImmutableMap.of("tag=1", TAG_VALUE_1, TAG_KEY_2, TAG_VALUE_2)))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessage("The tag key name: tag=1 contains the forbidden character: =");
+    }
+
+    @Test
+    public void shouldThrowIfFirstMetricTagValueContainsEquals() throws Exception {
+        assertThatThrownBy(() -> MetricNameUtils.getMetricName(METRIC_NAME,
+                ImmutableMap.of(TAG_KEY_1, "tag=Val1", TAG_KEY_2, TAG_VALUE_2)))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessage("The tag value name: tag=Val1 contains the forbidden character: =");
+    }
+
+    @Test
+    public void shouldThrowIfSecondMetricTagValueContainsEquals() throws Exception {
+        assertThatThrownBy(() -> MetricNameUtils.getMetricName(METRIC_NAME,
+                ImmutableMap.of(TAG_KEY_1, TAG_VALUE_1, TAG_KEY_2, "tagVal=2")))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessage("The tag value name: tagVal=2 contains the forbidden character: =");
+    }
+
+    @Test
+    public void shouldThrowForFirstEncounteredErrorMultipleMetricArgsContainInvalidCharacters() throws Exception {
+        assertThatThrownBy(() -> MetricNameUtils.getMetricName("metric;Name",
+                ImmutableMap.of("tag;1", TAG_VALUE_1, TAG_KEY_2, "tagVal=2")))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessage("The metric name: metric;Name contains the forbidden character: ;");
+    }
+
+
+    @Test
+    public void shouldReturnANameIfTryingToRegisterMetricWithExactlyTenTags() throws Exception {
+        Map<String, String> tags = IntStream.range(0, 10)
+                .boxed()
+                .collect(Collectors.toMap(i -> "tag" + i, i -> "tagVal" + i));
+        assertThat(tags.size()).isEqualTo(10);
+        assertThat(MetricNameUtils.getMetricName(METRIC_NAME, tags)).contains(METRIC_NAME);
+    }
+
+    @Test
+    public void shouldThrowIfTryingToRegisterMetricWithMoreThanTenTags() throws Exception {
+        Map<String, String> tags = IntStream.range(0, 11)
+                .boxed()
+                .collect(Collectors.toMap(i -> "tag" + i, i -> "tagVal" + i));
+        assertThat(tags.size()).isEqualTo(11);
+        assertThatThrownBy(() -> MetricNameUtils.getMetricName(METRIC_NAME, tags))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessage("Too many tags set on the metric metricName. "
+                        + "Maximum allowed number of tags is 10, found 11.");
+    }
```

Comment:
Should these last two tests reference the constant limit from `MetricNameUtils`?